### ISSUE TITLE
Is it possible to keep a chunk of continuous gpu memory (e.g. 20G) in DDP mode for gradient synchronization？

### ISSUE THREAD

**Issue by lutao2021** [pytorch/pytorch#141813]
### 🚀 The feature, motivation and pitch

My training code will produce a large amount of small memory fragments. Therefore, in DDP mode, it usually reports that it can’t allocate 15.00GB memory in the backward stage.

I set **gradient_as_bucket_view=True** and it works well. However, when I use **gradient accumulation**, this method can’t work.

So I want to create a chunk of continuous GPU memory for gradient backward. Are there any methods that can achieve this goal?

cc @H-Huang @awgu @kwen2501 @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @c-p-i-o

**Comment by awgu** [comment]
It will not be easy to support this feature request. Perhaps, you can try to use expandable segments?

Use this env var:
```
PYTORCH_CUDA_ALLOC_CONF="expandable_segments:True"
```

**Comment by fegin** [comment]
With `gradient accumulation`, there will be at least 2 copies of the gradients. Can you verify the memory is actually enough?

**Comment by lutao2021** [comment]
> With `gradient accumulation`, there will be at least 2 copies of the gradients. Can you verify the memory is actually enough?

Yes, maybe `gradient accumulation` occupies some extra gpu memory. We slightly adjust the network to reduce the gpu memory occupancy. Now, it works. Thank you!
